frontend/src/store/projectProvider.tsx

```diff
@@ -0,0 +1,56 @@
+import { useReducer } from "react";
+import ProjectContext from "./project-context";
+
+const defaultProjectState = {
+  projects: [],
+  searchResults: [],
+};
+const projectReducer = (state: any, action: any) => {
+  if (action.type === "STORE_PROJECTS") {
+    return {
+      projects: action.projects,
```

Comment:
nit: i would recommend copying the project array so that changes to the `action.projects` array doesn't affect the reducer state

```
return {
  projects: [...action.projects],
  searchResults: state.searchResults,
}
```